mountpoint-s3-client/src/endpoint_config.rs

```diff
@@ -148,10 +182,59 @@ impl EndpointConfig {
         let resolved_endpoint = endpoint_rule_engine
             .resolve(endpoint_request_context)
             .map_err(EndpointError::UnresolvedEndpoint)?;
-        let endpoint_uri = resolved_endpoint.get_url();
-        Uri::new_from_str(&allocator, endpoint_uri)
+
+        Ok(ResolvedEndpointInfo(resolved_endpoint))
+    }
+}
+
+/// Wrapper for [ResolvedEndpoint] from CRT to get [Uri] and [AuthScheme]
+#[derive(Debug, Clone)]
```

Comment:
Maybe I'm not looking hard enough, but I don't see anywhere that needs this to be `Clone`, which also means you don't need `ResolvedEndpoint` to be `Clone`.